lib/json/truffle_ruby/generator.rb

```diff
@@ -303,7 +303,7 @@ def to_h
         # GeneratorError exception.
         def generate(obj, anIO = nil)
           if @indent.empty? and @space.empty? and @space_before.empty? and @object_nl.empty? and @array_nl.empty? and
-              !@ascii_only and !@script_safe and @max_nesting == 0 and !@strict
```

Comment:
I think we still need this check, the `else` of the `case` doesn't check for `@strict`